dlt/resource_dlt_pipeline.go

```diff
@@ -0,0 +1,357 @@
+	return s
+}
+
+func (Pipeline) CustomizeSchema(s *common.CustomizableSchema) *common.CustomizableSchema {
+	// Required fields
+	// s.SchemaPath("name").SetRequired()
+	// s.SchemaPath("storage").SetRequired()
+	// s.SchemaPath("cluster").SetRequired()
+	// s.SchemaPath("continuous").SetRequired()
+
+	// ForceNew fields
+	s.SchemaPath("storage").SetForceNew()
+	s.SchemaPath("catalog").SetForceNew()
+
+	// Computed fields
+	s.SchemaPath("id").SetComputed()
+	s.SchemaPath("cluster", "node_type_id").SetComputed()
+	s.SchemaPath("cluster", "driver_node_type_id").SetComputed()
+	// s.SchemaPath("cluster", "enable_local_disk_encryption").SetComputed()
+	s.SchemaPath("url").SetComputed()
+
+	// SuppressDiff fields
+	s.SchemaPath("edition").SetSuppressDiff()
+	s.SchemaPath("channel").SetSuppressDiff()
+	s.SchemaPath("cluster", "spark_conf").SetCustomSuppressDiff(clusters.SparkConfDiffSuppressFunc)
+	s.SchemaPath("cluster", "aws_attributes", "zone_id").SetCustomSuppressDiff(clusters.ZoneDiffSuppress)
+	s.SchemaPath("cluster", "autoscale", "mode").SetCustomSuppressDiff(common.EqualFoldDiffSuppress)
+	s.SchemaPath("edition").SetCustomSuppressDiff(common.EqualFoldDiffSuppress)
+	s.SchemaPath("storage").SetCustomSuppressDiff(suppressStorageDiff)
+
+	// Deprecated fields
+	s.SchemaPath("cluster", "init_scripts", "dbfs").SetDeprecated(clusters.DbfsDeprecationWarning)
+
+	// Delete fields
+	s.SchemaPath("cluster", "gcp_attributes").RemoveField("use_preemptible_executors")
+	s.SchemaPath("cluster", "gcp_attributes").RemoveField("boot_disk_size")
+
+	// Default values
+	s.SchemaPath("edition").SetDefault("ADVANCED")
+	s.SchemaPath("channel").SetDefault("CURRENT")
+
+	// ConflictsWith fields
+	s.SchemaPath("storage").SetConflictsWith([]string{"catalog"})
+	s.SchemaPath("catalog").SetConflictsWith([]string{"storage"})
+
+	// MinItems fields
+	s.SchemaPath("library").SetMinItems(1)
+	s.SchemaPath("notification", "email_recipients").SetMinItems(1)
+	s.SchemaPath("notification", "alerts").SetMinItems(1)
+
+	// MaxItems fields
+	s.SchemaPath("cluster", "ssh_public_keys").SetMaxItems(10)
+	s.SchemaPath("cluster", "init_scripts").SetMaxItems(10)
+
+	// ValidateFunc fields
+	s.SchemaPath("channel").SetValidateFunc(validation.StringInSlice([]string{"current", "preview"}, true))
+	s.SchemaPath("edition").SetValidateFunc(validation.StringInSlice([]string{"pro", "core", "advanced"}, true))
+
+	return s
+}
+
+var pipelineSchema = common.StructToSchema(Pipeline{}, nil)
+
+type workspaceClient struct {
+	*databricks.WorkspaceClient
+}
+
+func ResourcePipeline() common.Resource {
+	return common.Resource{
+		Schema: pipelineSchema,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			wsClient := workspaceClient{w}
+			err = wsClient.Create(ctx, d, d.Timeout(schema.TimeoutCreate))
+			if err != nil {
+				return err
+			}
+
+			// why is it required
```

Comment:
I assume because it's useful for customers to have the url of a job (much more than the url of a users). Let's add a note.